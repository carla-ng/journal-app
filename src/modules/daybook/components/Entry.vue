<template>
  <div class="entry-container pointer p-3">

      <div class="entry-title d-flex">
          <span class="text-success fs-5 fw-bold">15</span>
          <span class="mx-1 fs-5">July</span>
          <span class="mx-1 fs-5 fw-light">2022, Thursday</span>
      </div>

      <div class="entry-description">
            Nostrud ex veniam anim occaecat sint amet irure cupidatat enim cupidatat. Laboris magna irure nostrud duis elit eiusmod aute proident id. Tempor ullamco velit exercitation quis labore amet mollit irure nulla magna. Ipsum consectetur in quis id labore ea. Aliquip consequat quis proident consectetur occaecat officia do do mollit duis ad laborum. Aute deserunt minim est duis pariatur elit dolor fugiat ea enim incididunt ullamco.
      </div>
  </div>
</template>

<script>
export default {

}
</script>

<style lang="scss" scoped>
.entry-container {
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;

    &:hover {
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
    }

    .entry-description {
        font-size: 12px;
    }
}
</style>