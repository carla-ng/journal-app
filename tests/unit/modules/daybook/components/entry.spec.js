import { shallowMount } from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry.vue'
import { journalState } from '../../../mock-data/test-journal-state'

describe('Entry Component tests', () => {

    const mockRouter = {
        push: jest.fn()
    }
    
    const wrapper = shallowMount( Entry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })

    test('Must match snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('Must redirect when clicking entry-container', () => {
        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')

        expect( mockRouter.push ).toHaveBeenCalledWith(
            { "name": "entry", "params": {"id": journalState.entries[0].id }}
        )
    })

    test('Tests with computed properties', () => {
        expect( wrapper.vm.day ).toBe(18)
        expect( wrapper.vm.month ).toBe('July')
        expect( wrapper.vm.yearAndDay ).toBe('2021, Sunday')
    })

})