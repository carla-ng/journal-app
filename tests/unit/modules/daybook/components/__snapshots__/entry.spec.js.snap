// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Component tests Must match snapshot 1`] = `
<div class="entry-container pointer p-3">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">18</span><span class="mx-1 fs-5">July</span><span class="mx-1 fs-5 fw-light">2021, Sunday</span></div>
  <div class="entry-description">Hi from mock data</div>
</div>
`;
